File: src/android/com/adobe/phonegap/push/notifications/channels/ICustomChannel.java
package com.adobe.phonegap.push.notifications.channels;

import android.app.Notification;
import android.app.NotificationManager;
import android.media.AudioAttributes;
import android.net.Uri;
import android.provider.Settings;

public interface ICustomChannel {
    String CHANNEL_ID = "";
    String PREFIX_ID = "";
    String name = "";
    Boolean vibration = false;
    long[] vibrationPattern = {};
    int visibility = Notification.VISIBILITY_PUBLIC;
    Uri sound = Settings.System.DEFAULT_RINGTONE_URI;
    AudioAttributes audioAttributes = null;
    int light = -1;
    Boolean showBadge = false;
    int priority = NotificationManager.IMPORTANCE_DEFAULT;
}

File: src/android/com/adobe/phonegap/push/notifications/channels/ICustomChannel.kt
package com.adobe.phonegap.push.notifications.channels

import android.app.Notification
import android.media.AudioAttributes
import android.app.NotificationManager
import android.provider.Settings

interface ICustomChannel {
    companion object {
        const val CHANNEL_ID = ""
        const val PREFIX_ID = ""
        const val name = ""
        const val vibration = false
        val vibrationPattern = longArrayOf()
        const val visibility = Notification.VISIBILITY_PUBLIC
        val sound = Settings.System.DEFAULT_RINGTONE_URI
        val audioAttributes: AudioAttributes? = null
        const val light = -1
        const val showBadge = false
        const val priority = NotificationManager.IMPORTANCE_DEFAULT
    }
}